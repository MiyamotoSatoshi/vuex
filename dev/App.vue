<template>
  <playground></playground>
</template>

<script>
  import Playground from './Playground'

  export default {
    components: {
      Playground
    }
  }
</script>
