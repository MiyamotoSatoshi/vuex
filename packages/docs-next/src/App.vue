<template>
  <v-fade-transition mode="out-in">
    <router-view />
  </v-fade-transition>
</template>

<script>
  export default {
    name: 'App',

    metaInfo: {
      title: 'Welcome to Vuetify',
      titleTemplate: '%s | Vuetify.js',
    },
  }
</script>
