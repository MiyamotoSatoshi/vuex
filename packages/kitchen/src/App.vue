<template>
  <v-app :dark="$vuetify.dark">
    <v-toolbar
      color="blue-grey"
      dark
      fixed
      app
      clipped-right
      dir="ltr"
    >
      <v-toolbar-side-icon @click.stop="drawer = !drawer" />
      <v-toolbar-title class="white--text headline">
        üç£
        <span class="font-weight-black">
          VUETIFY
        </span>
        <span class="font-weight-light">
          KITCHEN
        </span>
      </v-toolbar-title>
      <v-spacer />
      <codepen />
      <v-btn
        icon
        title="Change theme"
        @click="$vuetify.dark = !$vuetify.dark"
      >
        <v-icon>
          mdi-invert-colors
        </v-icon>
      </v-btn>
      <v-btn
        icon
        title="Toggle RTL"
        @click="$vuetify.rtl = !$vuetify.rtl"
      >
        <v-icon v-if="$vuetify.rtl">
          mdi-format-horizontal-align-right
        </v-icon>
        <v-icon v-else>
          mdi-format-horizontal-align-left
        </v-icon>
      </v-btn>
      <v-btn
        v-if="$route.params.component"
        :href="href"
        target="_blank"
        rel="noopener"
        icon
        title="Edit this page on Github"
      >
        <v-icon>mdi-pencil</v-icon>
      </v-btn>
    </v-toolbar>
    <v-navigation-drawer
      v-model="drawer"
      app
      fixed
    >
      <v-list
        dense
        class="pt-0"
      >
        <v-list-tile
          v-for="item in items"
          :key="item.title"
          :to="item.to"
        >
          <v-list-tile-content>
            <v-list-tile-title>{{ item.title }}</v-list-tile-title>
          </v-list-tile-content>
        </v-list-tile>
      </v-list>
    </v-navigation-drawer>
    <router-view />
  </v-app>
</template>

<script>
  import meals from './pan'

  const items = meals.map(meal => ({ title: meal, to: meal }))

  export default {
    components: {
      Codepen: () => import('@/components/Codepen')
    },
    filters: {
      capitalize (value) {
        if (!value) return ''
        value = value.toString()
        return value.charAt(0).toUpperCase() + value.slice(1)
      }
    },
    data: () => ({
      items: [
        { title: 'Home', to: '/' },
        { title: 'Custom theme config', to: '/theme' },
        ...items
      ],
      drawer: false
    }),
    computed: {
      href () {
        return `https://github.com/vuetifyjs/vuetify/tree/next/packages/kitchen/src/pan/${this.$route.params.component}.vue` // TODO: set branch back to master before v2.0 release
      }
    }
  }
</script>
