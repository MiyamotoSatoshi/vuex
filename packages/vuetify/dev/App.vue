<template>
  <v-app>
    <v-content>
      <playground></playground>
    </v-content>
  </v-app>
</template>

<script>
  import Playground from './Playground'

  export default {
    components: {
      Playground
    }
  }
</script>
