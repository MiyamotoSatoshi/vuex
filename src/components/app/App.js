import Themeable from '~mixins/themeable'

export default {
  functional: true,

  mixins: [Themeable],

  props: {
    fixedFooter: Boolean,
    footer: Boolean,
    id: {
      type: String,
      default: 'app'
    },
    toolbar: Boolean
  },

  render (h, { props, data, children }) {
    data.staticClass = (`application ${data.staticClass || ''}`).trim()
    data.staticClass += ` application--${props.dark ? 'dark' : 'light'}`

    props.footer && (data.staticClass += ' application--footer')
    props.fixedFooter && (data.staticClass += ' application--footer-fixed')
    props.toolbar && (data.staticClass += ' application--toolbar')

    data.attrs = { 'data-app': true }
    data.domProps = { id: props.id }

    return h('div', data, children)
  }
}
