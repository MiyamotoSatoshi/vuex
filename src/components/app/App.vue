<template lang="pug">
  div(
    class="with"
    v-bind:id="id"
    v-bind:class="classes"
    data-app
  )
    slot
</template>

<script>
  export default {
    props: {
      footer: Boolean,

      leftFixedSidebar: Boolean,

      leftSidebar: Boolean,

      id: {
        type: String,
        default: 'app'
      },

      rightFixedSidebar: Boolean,

      rightSidebar: Boolean,

      topFixedToolbar: Boolean,

      topToolbar: Boolean,

      sidebarUnderToolbar: Boolean
    },

    computed: {
      classes () {
        return {
          'left-fixed-sidebar': this.leftFixedSidebar,
          'left-sidebar': this.leftSidebar,
          'bottom-footer': this.footer,
          'right-fixed-sidebar': this.rightFixedSidebar,
          'right-sidebar': this.rightSidebar,
          'top-fixed-toolbar': this.topFixedToolbar,
          'top-toolbar': this.topToolbar,
          'sidebar-under-toolbar': this.sidebarUnderToolbar
        }
      }
    }
  }
</script>