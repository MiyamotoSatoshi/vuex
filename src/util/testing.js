import Vue from 'vue'
import load from '~util/load'
import { mount, shallow } from 'avoriaz'

export function test(name, cb) {
  Vue.prototype.$vuetify = { load }

  beforeEach(() => {
    rafPolyfill(window)
  })

  describe(name, () => cb({
    functionalContext,
    mount,
    shallow
  }))
}

export function functionalContext(context = {}) {
  return {
    context: Object.assign({
      data: {},
      props: {}
    }, context)
  }
}


//requestAnimationFrame polyfill | Milos Djakonovic ( @Miloshio ) | MIT | https://github.com/milosdjakonovic/requestAnimationFrame-polyfill
export function rafPolyfill(w) {
  /**
   *
   * How many times should polyfill call
   * update callback? By canon, it should
   * be 60 times per second, so that ideal
   * framerate 60fps could be reached.
   *
   * However, even native implementations
   * of requestAnimationFrame often cannot
   * do 60fps, but, unlike any polyfill,
   * they can synchronise achievable fps
   * rate with screen refresh rate.
   *
   * So, leave this value 1000/60 unless
   * you target specific browser on spec
   * ific device that is going to work
   * better with custom value. I think
   * that this is the longest comment I've
   * written on single variable so far.
  **/
  var FRAME_RATE_INTERVAL = 1000/60,

  /**
   * All queued callbacks in given cycle.
  **/
  allCallbacks = [],

  executeAllScheduled = false,

  shouldCheckCancelRaf = false,

  /**
   * Callbacks queued for cancellation.
  **/
  callbacksForCancellation = [],

  /**
   * Should callback be cancelled?
   * @param cb - callback
  **/
  isToBeCancelled = function(cb){
    for(var i=0;i<callbacksForCancellation.length;i++){
      if(callbacksForCancellation[i] === cb ){
        callbacksForCancellation.splice(i,1);
        return true;
      }
    }
  },



  /**
   *
   * Executes all (surprise) callbacks in
   * and removes them from callback queue.
   *
  **/
  executeAll = function(){
    executeAllScheduled = false;
    var _allCallbacks = allCallbacks;
    allCallbacks = [];
    for(var i=0;i<_allCallbacks.length;i++){
      if(shouldCheckCancelRaf===true){
        if (isToBeCancelled(_allCallbacks[i])){
          shouldCheckCancelRaf = false;
          return;
        }
      }
      _allCallbacks[i].apply(w, [ new Date().getTime() ] );
    }
  },

  /**
   *
   * requestAnimationFrame polyfill
   * @param callback - callback to be queued & executed | executed
   * @return callback
   *
  **/
  raf = function(callback){
    allCallbacks.push(callback);
    if(executeAllScheduled===false){
      w.setTimeout(executeAll, FRAME_RATE_INTERVAL);
      executeAllScheduled = true;
    }
    return callback;
  },

  /**
   *
   * Cancels raf.
  **/
  cancelRaf = function(callback){
    callbacksForCancellation.push(callback);
    shouldCheckCancelRaf = true;
  },


  //https://gist.github.com/paulirish/1579671
  vendors = ['ms', 'moz', 'webkit', 'o'];

    for(var x = 0; x < vendors.length && !w.requestAnimationFrame; ++x) {
        w.requestAnimationFrame = w[vendors[x]+'RequestAnimationFrame'];
        w.cancelAnimationFrame = w[vendors[x]+'CancelAnimationFrame']
        || w[vendors[x]+'CancelRequestAnimationFrame'];
    }

  if (!w.requestAnimationFrame) w.requestAnimationFrame = raf;
  if (!w.cancelAnimationFrame)  w.cancelAnimationFrame  = cancelRaf;

}
