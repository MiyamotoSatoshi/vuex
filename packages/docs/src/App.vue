<template>
  <v-fade-transition mode="out-in">
    <router-view />
  </v-fade-transition>
</template>

<script>
  // Mixins
  import Meta from '@/mixins/meta'

  // Utilities
  import { waitForReadystate } from '@/util/helpers'

  // Data
  import languages from '@/data/i18n/languages.json'

  const fallbackLocale = languages.find(lang => lang.fallback === true).locale

  export default {
    name: 'App',

    mixins: [Meta],

    data: () => ({
      availableLocales: languages.map(lang => lang.alternate || lang.locale),
      languages,
    }),

    computed: {
      languageIsValid () {
        return this.availableLocales.includes(this.$route.params.lang)
      },
    },

    watch: {
      '$route.path' () {
        typeof window !== 'undefined' && window.scrollTo(0, 0)
      },
    },

    created () {
      if (!this.languageIsValid) this.$router.push(`/${fallbackLocale}`)

      if (this.$ssrContext) return

      this.$vuetify.theme.dark = localStorage.getItem('vuetify__documentation__theme') === 'true'
    },

    async mounted () {
      if (!this.$route.hash) return

      await this.$nextTick()
      await waitForReadystate()

      this.$vuetify.goTo(this.$route.hash)
    },
  }
</script>

<style lang="sass">
  .text-decoration-none
    text-decoration: none

  .wf-loading .material-icons
    display: none

  .v-application .markdown code
    box-shadow: none
    color: #c0341d
    background-color: #fbe5e1

  .v-application .markdown kbd > code
    background: transparent
    color: inherit
</style>
