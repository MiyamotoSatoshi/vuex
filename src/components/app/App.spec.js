import App from '~components/app/App'
import { test } from '~util/testing'

test('App.js', ({ mount, functionalContext }) => {
  it('should have an application class', () => {
    const wrapper = mount(App, functionalContext())

    expect(wrapper.hasClass('application')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have a toolbar and footer class', () => {
    const wrapper = mount(App, functionalContext({
      props: {
        toolbar: true,
        footer: true,
        fixedFooter: true
      }
    }))

    expect(wrapper.hasClass('application--toolbar')).toBe(true)
    expect(wrapper.hasClass('application--footer')).toBe(true)
    expect(wrapper.hasClass('application--footer-fixed')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have data-app attribute', () => {
    const wrapper = mount(App, functionalContext())

    expect(wrapper.hasAttribute('data-app', 'true')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should allow a custom id', () => {
    const wrapper = mount(App, functionalContext({
      props: {
        id: 'inspire'
      }
    }))

    expect(wrapper.hasAttribute('id', 'inspire')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
