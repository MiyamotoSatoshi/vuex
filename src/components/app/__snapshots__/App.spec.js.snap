// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js should allow a custom id 1`] = `

<div data-app="true"
     class="application application--light"
     id="inspire"
>
</div>

`;

exports[`App.js should have a toolbar and footer class 1`] = `

<div data-app="true"
     class="application application--light application--footer application--footer-fixed application--toolbar"
     id="app"
>
</div>

`;

exports[`App.js should have an application class 1`] = `

<div data-app="true"
     class="application application--light"
     id="app"
>
</div>

`;

exports[`App.js should have data-app attribute 1`] = `

<div data-app="true"
     class="application application--light"
     id="app"
>
</div>

`;
